### ISSUE TITLE
IINA doesn't open when SMB sharing is turned off 

### ISSUE THREAD

**Issue by parthokr** [iina/iina#5067]
**System and IINA version:**

- macOS 14 
- IINA 1.3.5

**Expected behavior:**
IINA should load up regardless of any previously known network share.

**Actual behavior:**
IINA keeps on loading.

**Steps to reproduce:**
* Mount any SMB location and play some media via IINA
* Turn off SMB share
* Try opening IINA with some local file
* IINA ends up banging itself on the dock 

**How often does this happen?**
Every time until it is forced to quit several times.

**Comment by ShlomoCode** [comment]
Duplicate of https://github.com/iina/iina/issues/4973?

**Comment by low-batt** [comment]
Yes. Very likely duplicate of #4973. [This comment](https://github.com/iina/iina/issues/4973#issuecomment-2181352809) contains the workaround to get IINA to startup.

The offending code is not supposed to kick in unless the recent documents list is empty and the copy of the list of recent documents stored in [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) under the key `recentDocuments` is not. If you clear the recent documents using `Clear Menu` under the `File` -> `Open Recent` menu then IINA also clears `recentDocuments`. Now if you fall back to IINA 1.3.4 and click `Clear Menu` then [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) will be out of sync. The expected way to end up with no entries in `Open Recent` and entries still in [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults)  `recentDocuments` is the situation this workaround attempted to address. As of Sonoma macOS will clear the list of recent documents in `Open Recent` if an unsigned version of IINA is run, such as a nightly build.

I'm worried there is some other way the list of recent documents is being cleared that we have missed.

The plan is to disable this workaround in the next release of IINA. Developers who have to run unsigned builds all the time will still be able to enable it using a [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) key.
